File: Seriophiles/SeriophilesSPA/app/src/app/shows/domain/models/tvShow.ts
import { Genre } from "./genres.enum";
import {IEpisode} from "./episode";

interface IRating {
  average: number
}

export interface ITvShow{
  id: number,
  name: string,
  year: number,
  genres: Genre[],
  runtime: number,
  premiered: string,
  ended?: number,
  site?: string,
  language?: string,
  rating?: IRating,
  summary?: string,
  episodes: IEpisode[],
  cast: {[key: string]: string}
}

File: Seriophiles/Services/Series/Series.API/Entities/TVShow.cs
using Series.API.DTOs;
namespace Series.API.Entities
{
    public class TVShow
    {
        public int id { get; set; }
        public string? name { get; set; }
        public string[]? genres { get; set; }
        public int? year { get; set; }
        public int? runtime { get; set; }
        public string? premiered { get; set; }
        public string? ended { get; set; }
        public string? site { get; set; }
        public string? language { get; set; }
        public List<SeasonDTO>? seasons { get; set; }
        public List<CharacterDTO>? cast { get; set; }
        public string? summary { get; set; }
        public string? image { get; set; }
    }
}
